util.go

```diff
@@ -74,15 +74,13 @@ func schemeFromURL(url string) (string, error) {
 		return "", errEmptyURL
 	}
 
-	u, err := nurl.Parse(url)
-	if err != nil {
-		return "", err
-	}
-	if len(u.Scheme) == 0 {
+	parts := strings.SplitN(url, "://", 2)
```

Comment:
Looking for `://` feels safer to me; also, this will panic if `url` doesn't contain any `:`

Reply:
> Looking for `://` feels safer (and more accurate) to me;

Using `://` is neither safer nor more accurate as `:` is not a [valid scheme character](https://tools.ietf.org/html/rfc3986#section-3.1).

>  this will panic if `url` doesn't contain any `:`

That's what I meant by "with proper error handling"

Reply:
> Using :// is neither safer nor more accurate as : is not a valid scheme character.

You're right, the `//` is not required if there's no authority (https://tools.ietf.org/html/rfc3986#section-3); is that an actual situation though for DSN's?

> That's what I meant by "with proper error handling"

My worry was that we tried to save some bytes, and in the end, end up with re-implementing `strings.SplitN`